test/assertions/to-have-a-value-satisfying.spec.js

```diff
@@ -96,11 +98,13 @@ describe('to have a value satisfying assertion', () => {
       'to be a string'
     );
 
-    expect({ foo: '0', bar: '1' }, 'not to have a value satisfying', function(
-      value
-    ) {
-      expect(value, 'to be a number');
-    });
+    expect(
+      { foo: '0', bar: '1' },
+      'not to have a value satisfying',
+      expect.it(function(value) {
+        expect(value, 'to be a number');
+      })
```

Comment:
Why did we convert to the string version above and not here?

```js
expect(
   { foo: '0', bar: '1' },
   'not to have a value satisfying',
   'to be a number'
});
```

Reply:
No good reason, it seems like the idea behind these tests is to get wide coverage, so I changed the other occurrence back in 8446402a6359208fdb33aae23d1e348e518e2761 now.

Reply:
Fair.

Reply:
I know it is not you who added these things initially (me). But I think it would be preferable to have tests that uses the idiomatic syntax and have special tests for for advanced cases separately. This is more a note to self though.